File: cpp/Array/L_minDiffBwLarAndSmall.cpp
#include <bits/stdc++.h>
using namespace std;

int minDifference(vector<int> &nums)
{
    if (nums.size() <= 4)
    {
        return 0;
    }
    sort(nums.begin(), nums.end());
    int k = nums.size() - 3;
    int ans = nums.back() - nums[0];
    for (int i = k - 1; i < nums.size(); i++)
    {
        ans = min(ans, nums[i] - nums[i - k + 1]);
    }
    return ans;
}

File: JavaScript/Array/L_minDiffBwLarAndSmall.js
var minDifference = function (nums) {
    if (nums.length <= 4) {
        return 0;
    }
    nums.sort((a, b) => a - b);
    let k = nums.length - 3;
    let ans = nums[nums.length - 1] - nums[0];
    for (let i = k - 1; i < nums.length; i++) {
        ans = Math.min(ans, nums[i] - nums[i - k + 1]);
    }
    return ans;
};
